src/url-utils.ts

```diff
@@ -30,6 +30,19 @@ const sharedRelativeUrlProperties =
 export type UrlString = string;
 
 /**
+ * Returns a URL with the basename removed from the pathname.  Strips the
+ * search off of the URL as well, since it will not apply.
+ */
+export function debaseUrl(href: UrlString): UrlString {
```

Comment:
debase doesn't seem like the right name here. 

It's used in few enough places that it seems worth giving it a longer, more descriptive name. removeSearchAndFilename?